flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/PartitionRequestClientFactory.java

```diff
@@ -131,6 +134,42 @@ void destroyPartitionRequestClient(ConnectionID connectionId, PartitionRequestCl
 		clients.remove(connectionId, client);
 	}
 
+	private NettyPartitionRequestClient connectChannelWithRetry(ConnectingChannel connectingChannel,
+										 ConnectionID connectionId, boolean needConnect)
+		throws IOException, InterruptedException {
+		int count = 0;
+		Exception exception = null;
+		do {
+			try {
+				if (needConnect) {
+					LOG.info("Connecting to {} at {} attempt", connectionId.getAddress(), count);
+					nettyClient.connect(connectionId.getAddress()).addListener(connectingChannel);
+				}
+
+				NettyPartitionRequestClient client = connectingChannel.waitForChannel();
+				clients.replace(connectionId, connectingChannel, client);
+				return client;
+			} catch (IOException | ChannelException e) {
+				LOG.error("Failed {} times to connect to {}", count, connectionId.getAddress(), e);
```

Comment:
I think it will print `0 times` for the 1st failure and `1 times` for the 2nd. Should  `count + 1` be used?

Reply:
Good catch. There is why I syncronized on connectionId in line 74. It makes only one thread can goes into the connection for particular connectionId, it either success or fail after several times of retries.